app/src/main/java/com/sschuraytz/charcoaldrawing/SaveDialogFragment.java

```diff
@@ -0,0 +1,47 @@
+package com.sschuraytz.charcoaldrawing;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+public class SaveDialogFragment extends DialogFragment {
+
+    SaveDrawing saveDrawing;
+    Context baseContext;
+    Activity baseActivity;
+    Bitmap bitmapToSave;
+
+    SaveDialogFragment(Context context, Activity activity, Bitmap currentBitmap) {
```

Comment:
Fragments are strange things. 
Personally I hate them.

Fragments should only have a default constructor (no arguments) because of some edge cases with the Android pause/resume system.